life_expectancy/cleaning.py

```diff
@@ -37,41 +26,36 @@ def clean_data(df: pd.DataFrame, country: str) -> pd.DataFrame:
     df_eu_life_expectancy = df_eu_life_expectancy.melt(id_vars=["unit", "sex", "age", "region"], var_name="year")
 
     # ensure data types
-    df_eu_life_expectancy["year"] = df_eu_life_expectancy["year"].astype(int)
+    df_eu_life_expectancy["year"] = df_eu_life_expectancy["year"].astype(np.int64)
     df_eu_life_expectancy["value"] = df_eu_life_expectancy["value"].str.extract(r"(\d+\.\d)")
     df_eu_life_expectancy["value"] = df_eu_life_expectancy["value"].astype(float)
 
     # drop nulls
-    df_eu_life_expectancy = df_eu_life_expectancy.dropna()
+    df_clean_data = df_eu_life_expectancy.dropna().copy()
 
     # filter data
-    df_pt_life_expectancy = df_eu_life_expectancy[df_eu_life_expectancy["region"] == country]
-
-    return df_pt_life_expectancy
+    if country is not None:
+        df_clean_data = df_clean_data[df_clean_data["region"] == country]
 
-
-def save_data(df: pd.DataFrame, path: str):
-    """Saves data to csv specified by the given path.
-
-    Args:
-        df (pd.DataFrame): The dataframe to export to csv.
-        path (str): The location to save the data to.
-    """
-    df.to_csv(path, index=False)
+    return df_clean_data.reset_index(drop=True)
 
 
 if __name__ == "__main__":  # pragma: no cover
+    INPUT_DATA_PATH = "life_expectancy/data/eu_life_expectancy_raw.tsv"
     parser = argparse.ArgumentParser()
-    parser.add_argument("-c", "--country", help="Specify country.", type=str, default="PT")
+    parser.add_argument("-c", "--country", help="Specify country.", type=str, default=None)
+    parser.add_argument("-i", "--input", help="Specify file to clean.", type=str, default=INPUT_DATA_PATH)
 
     # args
     args = parser.parse_args()
     arg_country = args.country
+    arg_input = args.input
 
     # paths
-    INPUT_DATA_PATH = "life_expectancy/data/eu_life_expectancy_raw.tsv"
-    output_data_path = f"life_expectancy/data/{arg_country.lower()}_life_expectancy.csv"
+    output_data_path = (
+        f"life_expectancy/data/{arg_country.lower() if args.country is not None else 'eu'}_life_expectancy.csv"
```

Comment:
It's worth checking whether this won't save the file to an unexpected place in your machine (say, the root dir).
My recommendation is to always use a relative path to the current file (which you can access with pathlib.Path(__file__))

Reply:
That's something that I picked up recently and will definitely use the pathlib more often from now on :)